internal/csi/iam.go

```diff
@@ -0,0 +1,291 @@
+
+	// Creates policy
+	policyOutput, err := resources.svc.CreatePolicy(&iam.CreatePolicyInput{
+		Description:    aws.String("awe-k8s-tester generated policy for testing EC2"),
+		PolicyDocument: aws.String(policyDocument),
+		PolicyName:     aws.String(fmt.Sprintf("aws-k8s-tester-policy-%s", uniqueSuffix)),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to add role to create new policy(%v)\n", err)
+	}
+	resources.policy = &iamResource{
+		name: policyOutput.Policy.PolicyName,
+		arn:  policyOutput.Policy.Arn,
+	}
+	resources.lg.Info("created policy", zap.String("policy-name", *resources.policy.name))
+
+	// Creates role
+	roleOutput, err := resources.svc.CreateRole(&iam.CreateRoleInput{
+		AssumeRolePolicyDocument: aws.String(assumeRoleDocument),
+		RoleName:                 aws.String(fmt.Sprintf("aws-k8s-tester-role-%s", uniqueSuffix)),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new role (%v)\n", err)
+	}
+	resources.role = &iamResource{
+		name: roleOutput.Role.RoleName,
+		arn:  roleOutput.Role.Arn,
+	}
+	resources.lg.Info("created role", zap.String("role-name", *resources.role.name))
+
+	// Attaches role to policy
+	_, err = resources.svc.AttachRolePolicy(&iam.AttachRolePolicyInput{
+		PolicyArn: resources.policy.arn,
+		RoleName:  resources.role.name,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to attach role to policy (%v)\n", err)
+	}
+	resources.lg.Info("attached role to policy")
+
+	// Adds role to instance profile
+	_, err = resources.svc.AddRoleToInstanceProfile(&iam.AddRoleToInstanceProfileInput{
+		InstanceProfileName: resources.instanceProfile.name,
+		RoleName:            resources.role.name,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to add role to instance profile (%v)\n", err)
+	}
+	resources.lg.Info("attached role to instance policy")
+
+	// TODO: make separate method and change so that only prints that commands need to be run
+	fmt.Printf("\nManually delete profile, role, and policy:\n%s && %s && %s && %s && %s\n\n",
+		fmt.Sprintf("aws iam remove-role-from-instance-profile --instance-profile-name %s --role-name %s",
+			*resources.instanceProfile.name,
+			*resources.role.name),
+		fmt.Sprintf("aws iam detach-role-policy --role-name %s --policy-arn %s",
+			*resources.role.name,
+			*resources.policy.arn),
+		fmt.Sprintf("aws iam delete-role --role-name %s", *resources.role.name),
+		fmt.Sprintf("aws iam delete-policy --policy-arn %s", *resources.policy.arn),
+		fmt.Sprintf("aws iam delete-instance-profile --instance-profile-name %s", *resources.instanceProfile.name),
+	)
+
+	// Delay is needed to ensure that permissions have been propagated.
+	// See the section "Launching an Instance with an IAM Role" at
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
+	time.Sleep(10 * time.Second)
+
+	return resources, nil
+}
+
+func (resources *iamResources) GetInstanceProfileName() *string {
+	return resources.instanceProfile.name
```

Comment:
I am likely going to remove this when I move the related code from test.go to internal/csi